assignmentPR/src/parser.c

```diff
@@ -12,28 +12,372 @@
+    if(numString == NULL){
+        return false;
+    }
+
+    int strLen = strLength(numString);
+    int numSpaces = 0;
+
+    for(int i = strLen - 1; numString[i] == ' '; i--){
+        numSpaces++;
+        numString[i] = '\0';
+    }
+
+    if(numSpaces > 0){
+        return true;
+    } else {
+        return false;
+    }
+    return true;
+}
+
+/*
+ * @brief Function to add a leading 0 if there is no number in
+ *        front of ".", i.e. ".01"
+ *
+ * @param numString[] Null terminated string containing a floating
+ *        point number, newString[] is the updated string with leading zero.
+ *        If a zero is not needed the numString will not be changed
+ *
+ * @return new char array with added 0
+ */
+bool addLeadingZero(char *numString, char *new_string){
+    if(numString[0] != '.'){
+        return false;
+    } else {
+        strCat(new_string, numString);
+    }
+    return true;
+}
+
+/*
+ * @brief function to return string left of  "."
+ *
+ * @param numString[] Null terminated string containing a floating
+ *        point number, newString[] is placeholder where the Characteristic
+ *        will be stored into
+ *
+ * @return char array of characteristic
+ */
+char *getCharacteristic(char *numString, char *newString){
+    for(int i = 0; numString[i] != '.'; i++){
+        newString[i] = numString[i];
+    }
+    return newString;
+}
+
+/*
+ * @brief function to return string right of  "."
+ *
+ * @param numString[] Null terminated string containing a floating
+ *        point number, newString[] is placeholder where the Mantissa
+ *        will be stored into
+ *
+ * @return char array of mantissa
+ */
+char *getMantissa(char *numString, char *newString){
+    int numStringIndex = 0;
+    int newStringIndex = 0;
+    while(numString[numStringIndex] != '.'){
+        numStringIndex++;
+    }
+    for(int i = numStringIndex+1; numString[i] != '\0'; i++){
+        newString[newStringIndex] = numString[i];
+        newStringIndex++;
+    }
+    return newString;
+}
+
+/*
+ * @brief Func to return the length of the mantissa
+ *
+ * @param numString[] Null terminated string containing a floating
+ *        point number
+ *
+ * @return Int of mantissa length
+ */
+int getMantissaLength(char *numString){
+    int mantissaIndex = 0;
+    int length = 0;
+    while(numString[mantissaIndex] != '.'){
+        mantissaIndex++;
+    }
+    for(int i = mantissaIndex+1; numString[i] != '\0'; i++){
+        length++;
+    }
+    return length;
+}
+
+/*
+ * @brief Takes two strs and combines them, putting the source at the
+ *        end of the destination string
+ *
+ * @detail Ensure the destination str is big enough for the two str
+ *
+ * @param *dest Str that source will be combined into
+ *
+ * @param *source Str that will be put at the end of the dest
+ *
+ * @return Str containing dest and source combined
+ */
+char * strCat(char *dest, char *source){
+    char *copyPtr = dest + strLength(dest);
```

Comment:
This is nitpicky but try to keep the same formatting when initializing a pointer. Ex: keep all pointers initialized like: char * ptr;